### PR TITLE
fix loading old opt state by adding option to not inject lr in root optimizer

### PR DESCRIPTION
**Pull request by taufeeque9**

### CODE REVIEW

File: cleanba/cleanba_impala.py [hunk 1]
```diff
@@ -630,6 +635,8 @@ def frozen_schedule(
         return optimizer  # type: ignore
 
     # Inject learning rate schedule at the top level so we can just get it from .hyperparams and log it.
+    if dont_inject_lr:
+        return optimizer_with_learning_rate(args.learning_rate)
```
Comment (rhaps0dy) [inline cleanba/cleanba_impala.py:639]: Wouldn't this produce a static (not decreasing) learning rate? Seems wrong.

**Review by rhaps0dy — COMMENTED** [review]
What kind of old models are we talking about? Ones that were trained with the other LR hyperparameter structure, or the original ones?

File: cleanba/cleanba_impala.py (hunk 1 above)
Comment (rhaps0dy) [reply cleanba/cleanba_impala.py:639]: I think what you wnat to do is if `dont_inject_lr`, put the schedule function directly into this part of the code:

```
                    optax.inject_hyperparams(rmsprop_pytorch_style)(
                        learning_rate=learning_rate,
                        eps=args.rmsprop_eps,
                        decay=args.rmsprop_decay,
                    )
                    if args.optimizer == "rmsprop"
                    else (
                        optax.inject_hyperparams(optax.adam)(
                            learning_rate=learning_rate,
                            b1=args.adam_b1,
                            b2=args.rmsprop_decay,
                            eps=args.rmsprop_eps,
                            eps_root=0.0,
                        )
                    )

```

but I'm not 100% sure.

Comment (taufeeque9) [reply cleanba/cleanba_impala.py]: Have fixed this. Can you recheck if it looks good to you?

**Review by rhaps0dy — APPROVED** [review]
